### PR TITLE
Added new immutable params

### PR DESCRIPTION
**Pull request by mark-todd**

### CODE REVIEW

File: epioncho_ibm/simulation.py [hunk 1]
```diff
@@ -97,17 +97,15 @@ def __init__(
...
-        return self.state.params
+    def get_current_params(self) -> Params:
+        return self.state.get_params()
 
-    @params.setter
-    def params(self, params: Params):
-        self.state.params = params
-        self._derive_params()
+    @property
+    def immutable_params(self):
```
Comment (mkcinu) [inline epioncho_ibm/simulation.py:107]: How about making this "private" ?

Comment (mkcinu) [reply epioncho_ibm/simulation.py]: The reason I said this is I can imagine someone scratching his/her head and asking themselves which one should they use to see the parameters.

Comment (mark-todd) [reply epioncho_ibm/simulation.py]: Can't be done - sunder not allowed for properties I believe

Comment (mark-todd) [reply epioncho_ibm/simulation.py]: But if they want to access the immutables - I guess why not - it's clear what it is

Comment (mkcinu) [reply epioncho_ibm/simulation.py]: >Can't be done - sunder not allowed for properties I believe
Who said?

Comment (mark-todd) [reply epioncho_ibm/simulation.py]: Ok I think you're right on this! Just uploading change

**Pull request merged**